### PR TITLE
Add FluentSearch Implementation

### PR DESCRIPTION
**Pull request by williamw2**
# Pull Request

## 📖 Description

Adds an implementation for the FluentUI Search component.

Example:
image

### General

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [x] I have added a new component
- [ ] I have modified an existing component

### CODE REVIEW

**Comment by williamw2** [comment]
> To keep in sync with the other pages, please rename to `SearchPage.cs`

Done.

**Review by EisenbergEffect — APPROVED** [review]

**Review by vnbaaij — APPROVED** [review]
Did we check if all events/parameters from the web component are declared here as well?

File: examples/FluentUI.Demo.Shared/Pages/SearchPage.razor [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    </svg>
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/Pages/SearchPage.razor:41]: We can also, or maybe should, use a `<FluentIcon>` component here (by leveraging it's `Slot` parameter)

File: examples/FluentUI.Demo.Shared/Pages/SearchPage.razor [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/Pages/SearchPage.razor:49]: We can also, or maybe should, use a `<FluentIcon>` component here (by leveraging it's `Slot` parameter)

**Review by vnbaaij — APPROVED** [review]

**Comment by williamw2** [comment]
> Did we check if all events/parameters from the web component are declared here as well?

Yes. The component is essentially the same as the TextField but with some extra UI in the template.

**Pull request merged**